src/main/java/seedu/recipe/model/recipe/IngredientBuilder.java

```diff
@@ -0,0 +1,68 @@
+package seedu.recipe.model.recipe;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.recipe.commons.util.AppUtil.checkArgument;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.NAME_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.parse;
+
+import java.util.HashMap;
+import java.util.List;
+
+import seedu.recipe.logic.parser.Prefix;
+
+/**
+ * Represents a recipe's ingredient in the recipe book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidIngredient(String)}
+ */
+public class IngredientBuilder {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ingredients should follow this format: \n"
+            + "`-a {AMOUNT} -e {ESTIMATED AMOUNT} -n {NAME} `"
+            + "`-cn {COMMON NAME} [-r {REMARKS}] [-s {SUBSTITUTION}]\n"
+            + "i.e. `-a 1 oz. -n butter -r cubed -s margarine`";
```

Comment:
personally, i think the fields "estimated amount" and "common names" are redundant since we likely won't be implementing any features related to that

if we're keeping it, can i clarify whether amount and estimated amount are mutually exclusive flags? also, common name seems like it's optional, but it's not encased in square brackets here

Reply:
I amended the IngredientBuilder file to include the square brackets and `...` for optional and multiple fields, I think the copy that you reviewed was outdated.